Exercises/kokkoskernels/CGSolve/Begin/cgsolve.cpp

```diff
@@ -190,72 +190,119 @@ int main( int argc, char* argv[] )
 
   // Check sizes.
   checkSizes( N );
-  
+
   Kokkos::initialize( argc, argv );
   {
-    //typedef Kokkos::DefaultExecutionSpace::array_layout  Layout;
-    //typedef Kokkos::LayoutLeft   Layout;
-    typedef Kokkos::LayoutRight  Layout;
+    // Define scalar types based on availability of kokkos half precision support
+#if defined(KOKKOS_HALF_T_IS_FLOAT)
+    using ScalarType = Kokkos::Experimental::half_t;
+    using AccumulatorType = float;
+#else
+    using ScalarType = double;
+    using AccumulatorType = ScalarType;
+#endif // KOKKOS_HALF_T_IS_FLOAT
+
+    // Define view template argument types
+    using DeviceType = Kokkos::Cuda;
+    using ExecutionSpace = DeviceType::execution_space;
+    //using Layout = ExecutionSpace::array_layout;
+    //using Layout = Kokkos::LayoutLeft;
+    using Layout = Kokkos::LayoutRight;
+
+    // Define view types
+    using ViewVectorType = Kokkos::View<ScalarType*, Layout, DeviceType>;
+    using AccumulatorVectorType = Kokkos::View<AccumulatorType, Layout, DeviceType>;
+
+    // Declare timer products.
+    struct timeval begin, end, c0, c1;
 
-    typedef Kokkos::View<double*, Layout>  ViewVectorType;
+    // Defined scalars used in loop
+    ScalarType one   = 1.0;
```

Comment:
Would you consider using `constexpr` or at least `const` for compile-time constants?  It would be weird to let someone change the value of one, for instance.

Reply:
Yes.